src/main/java/org/simulator/io/CSVImporter.java

```diff
@@ -54,19 +48,63 @@
   private static final transient Logger logger = Logger.getLogger(CSVImporter.class.getName());
 
   /**
-   * The header of the file
+   * This method reads the data from the CSV file and converts it
+   * into the map with key as the column name and value as the array of amounts
+   * for the particular column.
+   * 
+   * @param pathname The path of the CSV file
+   * @return columnsMap -> LinkedListHashMap
    */
-  private String[] header;
+  private Map<String, double[]> readDataFromCSV(String pathname) throws IOException {
+    Map<String, double[]> columnsMap = new LinkedHashMap<>();
+    BufferedReader reader = new BufferedReader(new FileReader(pathname));
+    String line = reader.readLine();
+    List<String> lines = new LinkedList<String>();
+    String[] identifiers = line.split(",");
+    if (line != null) {
+      line = reader.readLine();
+      while ((line != null) && !line.isEmpty()) {
+        lines.add(line);
+        line = reader.readLine();
+      }
+
+      if (identifiers[0].equalsIgnoreCase("time")){
+        identifiers[0] = identifiers[0].toLowerCase();
+      }
+      for (String s : identifiers) {
+        columnsMap.put(s, new double[lines.size()]);
+      }
+
+      for (int i = 0; i < lines.size(); i++) {
+        String[] column = lines.get(i).split(",");
```

Comment:
So, if we add another method, then this split `String` will become a variable. As always, when reusing the same value, it is already better to create a variable for it.

Reply:
@draeger, do you here mean adding a method like this?

```
private String[] splitString(String s, String separator) {
    return s.split(separator);
}
```

Reply:
And with this method, making a variable for the split String like `private static final String SPLIT = ",";`?

Reply:
Actually, I was thinking about something like this:
```
private Map<String, double[]> readDataFromCSV(String pathname) throws IOException {
  return readDataFromCSV(pathname, ",");
}
```
So, we would essentially call another private method with an additional argument. Everything is the same, just the method body would be moved to that new method and the separating `String` becomes a variable (to be used at at least two positions).